kubernetes/client.go

```diff
@@ -77,10 +77,34 @@ func (client *K8SClient) GetToken() string {
 	return client.token
 }
 
+func SetUserIdentificationFromRemoteSecretUser(config *rest.Config, user *RemoteSecretUser) {
+	if len(user.User.Token) > 0 {
+		config.BearerToken = user.User.Token
+	}
+	exec := user.User.Exec
+	if exec != nil {
+		config.ExecProvider = &api.ExecConfig{
+			Command:            exec.Command,
+			Args:               exec.Args,
+			Env:                exec.Env,
+			APIVersion:         exec.APIVersion,
+			InstallHint:        cleanANSIEscapeCodes(exec.InstallHint),
+			ProvideClusterInfo: exec.ProvideClusterInfo,
+			InteractiveMode:    exec.InteractiveMode,
+		}
```

Comment:
We are going to want a log message to indicate an ExecProvider has been detected, at minimum for auditing purposes.

```suggestion
		}
		log.Debugf("Auth ExecProvider has been detected: cmd=[%v], args=%v", exec.Command, exec.Args)
```